sdk/eventhub/Azure.Messaging.EventHubs/src/Producer/EventDataBatch.cs

```diff
@@ -146,9 +146,13 @@ public sealed class EventDataBatch : IDisposable
         /// <returns><c>true</c> if the event was added; otherwise, <c>false</c>.</returns>
         ///
         /// <remarks>
-        ///   When an event is accepted into the batch, its content and state are frozen; any
-        ///   changes made to the event will not be reflected in the batch nor will any state
-        ///   transitions be reflected to the original instance.
+        ///   When an event is accepted into the batch, changes made to its properties
+        ///   will not be reflected in the batch nor will any state transitions be reflected
+        ///   to the original instance.
+        ///
+        ///   An important consideration to note is that any <see cref="ReadOnlyMemory{T}" /> or byte array
+        ///   passed to the event or a <see cref="BinaryData" /> instance associated with the event
+        ///   is referenced by the batch and must remain valid and unchanged until the batch is disposed.
```

Comment:
totally fine as is, but I wanted to offer an alternative in case you think it may be more clear
```suggestion
        ///   Note:  Any <see cref="ReadOnlyMemory{T}" />, byte array, or
        ///   <see cref="BinaryData" /> instance associated with the event and
        ///   referenced by the batch and must remain valid and unchanged until the batch is disposed.
```

Reply:
Yeah, that definitely reads better.   Going to apply locally so that I can hit both areas in one commit.